### DISCUSSION TITLE
Google Summer of Code 2023 application

### DISCUSSION THREAD

**Discussion by ag4ums** [eclipse-sw360/sw360#1818] [Ideas]
Hello all!

Sw360 is planning to apply for [Google Summer of Code](https://summerofcode.withgoogle.com) this year. In preparation to do so, we are creating this discussion to gather ideas for the GSoC-2023 projects.

More details here https://wiki.eclipse.org/Google_Summer_of_Code

Please feel free to drop any questions here (or start a new discussion) you have regarding the program, you want to submit a proposal idea, you want to be mentor in the program, etc.

While submitting a project idea, please tag it to be "Medium" or "Large" sized and who you'll prefer working on it "student" or "professional".
Some resources from Google:
- https://google.github.io/gsocguides/mentor/defining-a-project-ideas-list
- https://google.github.io/gsocguides/mentor/making-your-ideas-page

**Comment by mcjaeger** [comment]
Hi @ag4ums , would there be a page with / for  project ideas?  (I was only finding this: https://wiki.eclipse.org/Google_Summer_of_Code_2023_Ideas )

**Reply by ag4ums** [reply]
Hi @mcjaeger,  Created https://github.com/eclipse/sw360/wiki/Google-Summer-of-Code-Proposals-2023 GSoC Submission, and this page is mainly to collect and discuss the ideas.

**Comment by GMishx** [comment]
# Idea!
## Title: Support CycloneDX report
### Goal: Creating CycloneDX reports along with SPDX and other formats
CycloneDX is another popular format for generating SBOMs. Having SW360 import reports in CycloneDX along with SPDX will enable integration with more tools.

Links:
- https://github.com/CycloneDX/cyclonedx-core-java
- https://cyclonedx.org/

| Category | Rating |
| :-- | :-- |
| Low Hanging Fruit | ** |
| Risk/Exploratory | * |
| Fun/Periphial | ** |
| Core Development | * |
| Project Infrastructure | ** |
| Project size | Large |
| Preferred contributor | Student |

**Comment by ag4ums** [comment]
# Idea!
## Title: Rest API and UI Improvement for sw360 project
### Goal: Creating and extending existing endpoints to support sw360 New UI based on react. 
SW360 uses Liferay as a frontend, we are currently looking for a new UI for sw360 based on react 

Links:
- https://github.com/eclipse/sw360/issues/1826

| Category | Rating |
| :-- | :-- |
| Low Hanging Fruit | - |
| Risk/Exploratory | * |
| Fun/Periphial | *** |
| Core Development | *** |
| Project Infrastructure | ** |
| Project size | Large |
| Preferred contributor | Student |

**Comment by ag4ums** [comment]
# Idea!
## Title: Creating Project as a service in Sw360
### Goal: Idea is to create a project as a service which can then be used by multiple BU/Org for reusing common component repository 

| Category | Rating |
| :-- | :-- |
| Low Hanging Fruit | - |
| Risk/Exploratory | * |
| Fun/Periphial | *** |
| Core Development | *** |
| Project Infrastructure | ** |
| Project size | Large |
| Preferred contributor | Student |

**Comment by ag4ums** [comment]
# Idea!
## Title: Exporting and Importing releases from a server to another server
### Goal: Creating an importer and exporter service in sw360 to export the release information from one server in json that can be imported to another server

| Category | Rating |
| :-- | :-- |
| Low Hanging Fruit | - |
| Risk/Exploratory | * |
| Fun/Periphial | *** |
| Core Development | *** |
| Project Infrastructure | ** |
| Project size | Large |
| Preferred contributor | Student |

**Comment by ag4ums** [comment]
# Idea!
## Title: Improving automation with fossology
### Goal: Extending the existing sw360-fossology automation so that we can automate the scanning and uploading fossology generated report seamlessly.  

| Category | Rating |
| :-- | :-- |
| Low Hanging Fruit | - |
| Risk/Exploratory | * |
| Fun/Periphial | *** |
| Core Development | *** |
| Project Infrastructure | ** |
| Project size | Large |
| Preferred contributor | Student |

**Comment by ag4ums** [comment]
# Idea!
## Title: ORT Integration
### Goal: Extending the existing ORT integration with sw360 (https://github.com/eclipse/sw360/issues/1179 ).  

| Category | Rating |
| :-- | :-- |
| Low Hanging Fruit | - |
| Risk/Exploratory | * |
| Fun/Periphial | *** |
| Core Development | *** |
| Project Infrastructure | ** |
| Project size | Large |
| Preferred contributor | Student |

**Comment by arunazhakesan** [comment]
# Idea!
## Title: Dashboard Portlet for SW360 - Project & Instance Level
### Goal: Creating dashboard with dynamic visualizations of project level and overall Db level data in SW360

Links:
- 

| Category | Rating |
| :-- | :-- |
| Low Hanging Fruit | - |
| Risk/Exploratory | * |
| Fun/Periphial | **** |
| Core Development | *** |
| Project Infrastructure | ** |
| Project size | Large |
| Preferred contributor | Student |

**Comment by arunazhakesan** [comment]
# Idea!
## Title: Release Portlet for SW360
### Goal: Create release portlet to provide an independent portlet for searching, editing and updating component release information. 

Links:
- 

| Category | Rating |
| :-- | :-- |
| Low Hanging Fruit | - |
| Risk/Exploratory | * |
| Fun/Periphial | *** |
| Core Development | *** |
| Project Infrastructure | ** |
| Project size | Large |
| Preferred contributor | Student |
